pkg/manifests/manifests.go

```diff
@@ -2064,10 +2065,12 @@ func (f *Factory) MetricsServerDeployment(apiAuthSecretName string, kubeletCABun
 	// are rotated.
 	dep.Spec.Template.Annotations["monitoring.openshift.io/serving-ca-secret-hash"] = hashByteMap(servingCASecret.Data)
 
-	// Hash the metrics client cert and propagate it as an annotation to the
-	// deployment's pods to trigger a new rollout when the metrics client cert
+	// Hash the metrics server client cert and propagate it as an annotation to the
+	// deployment's pods to trigger a new rollout when the metrics server client cert
 	// is rotated.
-	dep.Spec.Template.Annotations["monitoring.openshift.io/metrics-client-cert-hash"] = hashByteMap(metricsClientCert.Data)
+	// TODO(slashpai): Remove this line release after metrics server is GA
+	dep.Spec.Template.Annotations["monitoring.openshift.io/metrics-client-certs-hash-"] = ""
```

Comment:
My bad, we're not using library-go's utils for Deployement resources yet. So no need for this. I'll give this a try locally.

Reply:
tried a cluster bot with https://github.com/openshift/cluster-monitoring-operator/pull/2268/commits/4dcd2e4e096c897276f9681ab2cac3cb8bb47b2e, all seem ok.

The `dial tcp 172.30.0.1:443: connect: connection refused` don't seem to be metrics-server related as I explain on the ticket.

Reply:
Tested with https://github.com/openshift/cluster-monitoring-operator/pull/2268/commits/4dcd2e4e096c897276f9681ab2cac3cb8bb47b2e on the cluster bot, all seem ok.

Got some "dial tcp 172.30.0.1:443: connect: connection refused", but as I explain on the ticket, they're not metrics-server specific.

We just need to check out the payload tests.

Reply:
I just realized that CMO never creates these secrets (the prometheus' one as well), it delegates to the CSR controller (I don't know if it's a good practic, but it's working for prometheus')

Apparently there is an issue there, the metrics-server CSR controller is not able to create the secret.